### DISCUSSION TITLE
[HW1] Exception이 발생해야만 하는 경우의 test case

### DISCUSSION THREAD

**Discussion by JustinKim98** [prosyslab-classroom/cs424-program-reasoning#406] [Q&A] [answered]
### Question

Coverage를 높이기 위해 테스트 케이스를 추가하는 중에 궁금한 점이 생겨서 질문드립니다.
메뉴얼(Readme.md) 대로라면 0으로 나누는 경우 `Division_by_zero`, 다른 undefined behavior에서는 `Invalid_argument`를 출력하게끔 되어 있습니다.

이렇게 Exception이 반드시 발생해야 하는 테스트 케이스는 어떻게 추가해야 하는지 궁금합니다 (실제로 exception이 발생하는지 확인하는 용도).
기존에 있는 방식과 같이 file.input, file.expected를 추가하는 방법으로는 확인할 수가 없다면 dune 파일에 기존 포맷과 다른 다른 형태의 테스트 케이스를 추가해야 할 것 같은데 괜찮을까요?

감사합니다

**Comment by spearo2** [comment, marked as answer]
네, 질문 감사합니다.

dune에 stderr를 읽도록 포멧을 변경하여 테스트 케이스를 추가하는것이 가능하지만, 60%의 커버리지를 달성하기 위해 꼭 exception과 관련된 테스트케이스를 추가하실 필요는 없습니다.

감사합니다.
